### PR TITLE
Story 1

### PR DESCRIPTION
**Pull request by AlicjaZub**

### CODE REVIEW

**Review by lucyferrabee — APPROVED** [review]
I can't find anything to change... Well done!

File: functions.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+function details(array $array): string
+{
+    $result = '';
+    foreach ($array as $details) {
+        if(is_Array($details)) {
```
Comment (nealioa) [inline functions.php]: Put a space between `if` and `(`.

File: functions.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            $result .= '<img src="images/' . $details['image'] . '" alt="' . $details['name'] . '">';
+            $result .= '<p><b> Latin name: </b>' . $details['latin_name'] . '</p>';
+            $result .= '<p><b> Hardiness: </b>' . $details['hardiness_type'] . '</p></div>';
+        } else {
+            return 'Invalid information!';
+        }
+    }
+    return $result;
+}
+//If I want to have images in a separate function:
```
Comment (nealioa) [inline functions.php]: Blocks of commented-out code are not usually checked in. If you don't need it, don't include it.

File: index.php [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+session_start();
+include_once 'config.php';
+include_once 'functions.php';
+
+$query = $db->prepare('SELECT `name`, `latin_name`, `hardiness_type`, `image` FROM `details` 
+    JOIN `hardiness_types` ON `details`.`hardiness`=`hardiness_types`.`id`');
+$query->execute();
+$plants = $query->fetchAll();
+
+//Checking if the query worked:
```
Comment (nealioa) [inline index.php]: Again, commented out code.
Useful while your writing and debugging your code, but should be removed before you put stuff into GitHub.

File: index.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    <title>Plants :)</title>
+</head>
+<body>
+    <div class="container">
+        <?php echo details($plants); ?>
+    </div>
+</body>
+</html>
+
+<!--A way to do it with for loop - don't do it this way, use foreach-->
```
Comment (nealioa) [inline index.php]: Commented out code!

File: style.css [hunk 5]
```diff
@@ -0,0 +1,30 @@
+.items {
+    /*border: red solid 2px;*/
+    width: 25%;
+    text-align: center;
+
```
Comment (nealioa) [inline style.css]: Unnecessary blank line.

File: test/functions.php [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+require '../functions.php';
+
+use PHPUnit\Framework\TestCase;
+
+class functions extends TestCase
+{
+    public function testDetails_success()
+    {
+        $input = [
+            'plant1'=>[
```
Comment (nealioa) [inline test/functions.php]: Put a space either side of `=>`. E.g. `'plant1' => [`

File: test/functions.php [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                'name'=> 'Prayer Plant',
+                'latin_name'=> 'Maranta leuconeura',
+                'hardiness_type' => 'Tender',
+                'image'=> 'prayer.jpg'],
+            'plant2'=> [
```
Comment (nealioa) [inline test/functions.php]: Space before `=>`.

**Review by nealioa — CHANGES_REQUESTED** [review]

File: ignore_me [hunk 8]
```diff
@@ -0,0 +1,28 @@
+INSERT INTO `details` (`name`, `latin_name`, `hardiness`, `image`) VALUES
```
Comment (luciavelasco) [inline ignore_me]: Remove from pull request, add the form to story2 branch if appropriate.

File: index.php [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+session_start();
```
Comment (luciavelasco) [inline index.php]: is this being used?

File: style.css [hunk 10]
```diff
@@ -0,0 +1,29 @@
+.items {
+    /*border: red solid 2px;*/
```
Comment (luciavelasco) [inline style.css]: remove comment

File: index.php [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+include_once 'config.php';
+include_once 'functions.php';
+
+$query = $db->prepare('SELECT `name`, `latin_name`, `hardiness_type`, `image` FROM `details`
```
Comment (luciavelasco) [inline index.php]: this should be a function - `getPlants` or something

File: config.php [hunk 12]
```diff
@@ -0,0 +1,4 @@
+<?php
+
+$db = new PDO('mysql:host=db; dbname=plants', 'root', 'password');
```
Comment (luciavelasco) [inline config.php]: these should be in a function `connectDb`

File: test/functions.php [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+        $input = [
+            'name' => 'Prayer Plant',
+            'latin_name' => 'Maranta leuconeura',
+            'hardiness_type' => 'Tender',
+            'image' => 'prayer.jpg',
+            'plant1' => [
+                'name' => 'Prayer Plant',
+                'latin_name' => 'Maranta leuconeura',
+                'hardiness_type' => 'Tender',
+                'image' => 'prayer.jpg']];
```
Comment (luciavelasco) [inline test/functions.php]: nitpicking, could you but these square brackets on a new line
```
    'image' => 'prayer.jpg'
  ]
];
```

File: test/functions.php [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+        $this->expectException(TypeError::class);
+        details($input);
+    }
+    public function testDetails_malformed_float()
+    {
+        $input = 3.333;
+        $this->expectException(TypeError::class);
+        details($input);
+    }
+    public function testDetails_malformed_bool()
```
Comment (luciavelasco) [inline test/functions.php:78]: Nice, I love how many malformed input tests you have, it makes my inner QA (quality assurance tester) happy! 😁

File: index.php [hunk 15]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+session_start();
+include_once 'config.php';
```
Comment (luciavelasco) [inline index.php]: Use `require` instead of `include` here and below

**Review by luciavelasco — CHANGES_REQUESTED** [review]
See comments and refer back to me

**Comment by luciavelasco** [comment]
This pull request should be from `story1` into `main` (`s1task1` into `story1` is for peer review). Don't worry, just let the trainer know tomorrow that you're aware of that.

File: test/functions.php (hunk 14 above)
Comment (AlicjaZub) [reply test/functions.php:78]: Yay :)

**Pull request merged**